### PR TITLE
Candidate Service

### PR DESCRIPTION
**Pull request by ArslanTem**

### CODE REVIEW

File: DAL/InternshipDbContext.cs [hunk 1]
```diff
@@ -12,7 +12,7 @@ namespace DAL
     public class InternshipDbContext : IdentityDbContext<User>
     {
         public InternshipDbContext(DbContextOptions options) : base(options) { }
-
+
```
Comment (DShcharbakou) [inline DAL/InternshipDbContext.cs:15]: ???

File: DAL/Repositories/CandidateRepository.cs [hunk 2]
```diff
@@ -4,11 +4,13 @@
...
 using DAL.Models;
+using Microsoft.EntityFrameworkCore;
 
 namespace DAL.Repositories
 {
-    internal class CandidateRepository : BaseRepository<Candidate>
+    internal class CandidateSet : BaseRepository<Candidate>
     {
-        public CandidateRepository(InternshipDbContext internshipDbContext) : base(internshipDbContext) { }
+        public CandidateSet(InternshipDbContext internshipDbContext) : base(internshipDbContext) { }
```
Comment (DShcharbakou) [inline DAL/Repositories/CandidateRepository.cs:13]: What is it?

File: DAL/Repositories/UnitOfWork.cs [hunk 3]
```diff
@@ -32,7 +32,7 @@ public BaseRepository<Candidate> Candidates
 
                 if (this.candidateRep == null)
                 {
-                    this.candidateRep = new CandidateRepository(internshipDbContext);
+                    this.candidateRep = new CandidateSet(internshipDbContext);
```
Comment (DShcharbakou) [inline DAL/Repositories/UnitOfWork.cs:35]: for what purpose did you rename it?

File: UI/Controllers/CandidatesController.cs [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+using BLL.Interfaces;
+using BLL.Services;
+
+namespace UI.Controllers
+{
+    [ApiController]
+    [Route("api/[controller]")]
+    public class CandidatesController : ControllerBase
+    {
+        private ICandidateService candidateService;
```
Comment (DShcharbakou) [inline UI/Controllers/CandidatesController.cs:21]: private readonly

Comment (DShcharbakou) [reply UI/Controllers/CandidatesController.cs:21]: where is the constructor with service injection? Do you test this code? This code doesn't work!

File: UI/Controllers/CandidatesController.cs [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+        [HttpGet]
+        public List<CandidateDTO> GetAll()
+        {
+            return (List<CandidateDTO>)candidateService.GetAllCandidates();
+        }
+        [HttpGet]
+        public CandidateDTO GetById(int id,[FromBody] CandidateDTO candidate)
+        {
+            return candidateService.GetCandidateById(id);
+        }
```
Comment (DShcharbakou) [inline UI/Controllers/CandidatesController.cs:32]: for what purposes do you need these methods?

File: UI/Controllers/CandidatesController.cs [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+        {
+            candidateService.AddCandidate(candidate);
+        }
+
+        // DELETE api/users/5
+        [HttpDelete("{id}")]
+        public void Delete(int id, [FromBody] CandidateDTO candidate)
+        {
+            candidateService.DeleteCandidate(id);
+        }
```
Comment (DShcharbakou) [inline UI/Controllers/CandidatesController.cs:44]: for what purposes do you need this method?

File: UI/Controllers/CandidatesController.cs [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+        {
+            return (List<CandidateDTO>)candidateService.GetAllCandidates();
+        }
+        [HttpGet]
+        public CandidateDTO GetById(int id,[FromBody] CandidateDTO candidate)
+        {
+            return candidateService.GetCandidateById(id);
+        }
+        [HttpPost("{id}")]
+        public void Save(int id, [FromBody] CandidateDTO candidate)
```
Comment (DShcharbakou) [inline UI/Controllers/CandidatesController.cs:34]: via this method you will register candidate to the sandbox. Why this method call "Save"?

File: UI/Controllers/CandidatesController.cs [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+        public List<CandidateDTO> GetAll()
+        {
+            return (List<CandidateDTO>)candidateService.GetAllCandidates();
+        }
+        [HttpGet]
+        public CandidateDTO GetById(int id,[FromBody] CandidateDTO candidate)
+        {
+            return candidateService.GetCandidateById(id);
+        }
+        [HttpPost("{id}")]
```
Comment (DShcharbakou) [inline UI/Controllers/CandidatesController.cs:33]: I suppose it should be sandboxId

File: UI/Controllers/CandidatesController.cs [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+        public void Save(int id, [FromBody] CandidateDTO candidate)
+        {
+            candidateService.AddCandidate(candidate);
```
Comment (DShcharbakou) [inline UI/Controllers/CandidatesController.cs:36]: where is second parameter for sandboxId? rename this method, for example candidateService.RegisterCandidate(id, candidateDto)

**Review by DShcharbakou — CHANGES_REQUESTED** [review]
Also you need update CandidateDto with necessary fields and CandidateService

File: DAL/InternshipDbContext.cs (hunk 1 above)
Comment (ArslanTem) [reply DAL/InternshipDbContext.cs:15]: ctrl+k+d

**Pull request closed without merging**